aten/src/ATen/native/TensorFactories.cpp

```diff
@@ -122,6 +122,9 @@ Tensor empty(
     IntArrayRef size,
     at::optional<DimnameList> names,
     const TensorOptions& options) {
+  if (!names.has_value()) {
```

Comment:
I think you can just `if (! names)`

Reply:
They're the same in terms of performance (operator::bool is the same as has_value) and I wanted to be more explicit about it.